fox_consumer/consumer.py

```diff
@@ -0,0 +1,197 @@
+                1. Assumption - no problems in rabbitmq (same as scraper…)
+            2. Lost connection (msec/sec)
+                1. Assumption - no problems in rabbitmq (same as scraper…)
+            3. Exchange not available
+                1. Assumption - no problems in rabbitmq (same as scraper…)
+                2. Change to not exist - the consumer will create it
+        2. Input problems
+            1. Data from not authorize source
+                1. How to verify it? - need to google
+                2. Since we have a dedicated exchange I think we should use an assumption it won’t happen
+                (same as we talk about the unnecessary routing key)
+            2. Data corrupted
+                1. verify json data
+                    1. We can use json.loads() which will throw “ValueError” (in try-catch)
+                    2. verify the json data using schema ("jsonschema" package)
+        3. Connection to postgres
+            1. Server is down
+                1. Assumption - no problems in postgres (same as rabbitmq…)
+            2. Lost connection
+                1. Verify every query using try-catch (if necessary - reestablish the connection)
+            3. Cannot update the table (lost connection before executing - before updating table)
+                1. Verify in the initialization that the table exist and create it if needed
+                2. Verify every query using try-catch (if necessary - reestablish the connection)
+        4. Postgres problems
+            1. DB not available (not exist)
+                1. Assumption - the DB should be ready before running the service
+            2. create table - if it already exist
+                1. check if table exist - if not create
+                2. The table won’t be deleted after the consumer has initialized
+            3. Duplicate data
+                1. Look for the primary key (img-id) in the DB before saving item
+
+    """
+    def __init__(self):
+        """
+        creation of FoxConsumer (using the 'with' statement)
+        """
+        self.schema = {
+                "$schema": "http://json-schema.org/draft-04/schema#",
+                "title": "FoxItem",
+                "description": "An item for Fox catalog",
+                "type": "object",
+                "properties": {
+                    "item_img_id": {"type": "string"},
+                    "item_main_category": {"type": "string"},
+                    "item_type": {"type": "string"},
+                    "item_name": {"type": "string"},
+                    "item_price": {"type": "number"},
+                },
+                "required": ["item_img_id", "item_main_category", "item_type", "item_name", "item_price"]
+            }
+
+        self.initialize_postgress_connection()
+        self.initialize_rabbitmq_connection()
+
+    def close_connections(self):
+        # TODO - Caduri - since it should run continuously, do we need this method?
```

Comment:
I did thought about it,
the only way I could and should close all the connection is in case i'll catch event/error (unknown problem or desire to shutdown the program).